src/sdk/python/rtdip_sdk/pipelines/sources/python/delta.py

```diff
@@ -0,0 +1,82 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ..interfaces import SourceInterface
+from ..._pipeline_utils.models import Libraries, SystemType
+from ..._pipeline_utils.constants import get_default_package
+import polars as pl
+from polars import LazyFrame
+
+class PythonDeltaSource(SourceInterface):
+    '''
+    The Python Delta Source is used to read data from a Delta table without using Apache Spark, returning a Polars LazyFrame
+
+    Args:
+        path (str): Path to the Delta table. Can be local or in S3/Azure storage
+        version (optional int): Specify the Delta table version to read from. Defaults to the latest version
+        storage_options (optional dict): Used to read from AWS/Azure storage. For AWS use format {"aws_access_key_id": "<>", "aws_secret_access_key":"<>"}. For Azure use format {"azure_storage_account_name": "<>", "azure_storage_account_key": "<>"}.
+        pyarrow_options (optional dict): Data Access and Efficiency options when reading from Delta. See [to_pyarrow_dataset](https://delta-io.github.io/delta-rs/python/api_reference.html#deltalake.table.DeltaTable.to_pyarrow_dataset){ target="_blank" }.
+        without_files (optional bool): If True loads the table without tracking files
+    '''
+    path: str
+    version: int
+    storage_options: dict
+    pyarrow_options: dict
+    without_files: bool
+
+    def __init__(self, path: str, version: int = None, storage_options: dict = None, pyarrow_options: dict = None, without_files: bool = False):
+        self.path = path
+        self.version = version
+        self.storage_options = storage_options
+        self.pyarrow_options = pyarrow_options
+        self.without_files = without_files
+        
+    @staticmethod
+    def system_type():
+        '''
+        Attributes:
+            SystemType (Environment): Requires PYTHON
+        '''            
+        return SystemType.PYTHON
+
+    @staticmethod
+    def libraries():
+        libraries = Libraries()
+        return libraries
+    
+    @staticmethod
+    def settings() -> dict:
+        return {}
+    
+    def pre_read_validation(self):
+        return True
+    
+    def post_read_validation(self):
+        return True
+    
+    def read_batch(self) -> LazyFrame:
+        '''
+        Reads data from a Delta table into a Polars LazyFrame
+        '''
+        without_files_dict = {"without_files": self.without_files}
+        lf = pl.scan_delta(source= self.path, version= self.version, storage_options= self.storage_options, delta_table_options= without_files_dict, pyarrow_options= self.pyarrow_options)
```

Comment:
is there a reason we don't offer delta_table_options in this component?

Reply:
From what I could see without_files was the only option you can have but I can change it

Reply:
This is from the polars code

Reply:
https://delta-io.github.io/delta-rs/python/api_reference.html#deltalake.table.DeltaTable